internal/contour/handler.go

```diff
@@ -189,23 +183,26 @@ func (e *EventHandler) onUpdate(op interface{}) bool {
 	case opAdd:
 		return e.builder.Source.Insert(op.obj)
 	case opUpdate:
-		if cmp.Equal(op.oldObj, op.newObj,
-			cmpopts.IgnoreFields(contour_api_v1.HTTPProxy{}, "Status"),
-			cmpopts.IgnoreFields(contour_api_v1alpha1.ExtensionService{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1beta1.GatewayClass{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1beta1.Gateway{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1beta1.HTTPRoute{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1alpha2.TLSRoute{}, "Status"),
-			cmpopts.IgnoreFields(gatewayapi_v1alpha2.GRPCRoute{}, "Status"),
-			cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion"),
-			cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ManagedFields"),
-		) {
-			e.WithField("op", "update").Debugf("%T skipping update, only status has changed", op.newObj)
-			return false
+		old, oldOk := op.oldObj.(client.Object)
+		new, newOk := op.newObj.(client.Object)
+		if oldOk && newOk {
+			equal, err := k8s.IsObjectEqual(old, new)
+			// Error is returned if there was no support for comparing equality of the specific object type.
+			// We can still process the object but it will be always considered as changed.
+			if err != nil {
+				e.WithError(err).WithField("op", "update").Errorf("%T error comparing objects", op.newObj)
+			}
+			if equal {
+				e.WithField("op", "update").Debugf("%T skipping update, only status has changed", op.newObj)
```

Comment:
It'd be nice to:
- log the namespace/name of the resource here as well, for debugging purposes
- adjust the message to be a little more general since it's not 100% accurate -- something like "skipping update, no changes to relevant fields"

Reply:
I've added this. 

I also noticed that `obj.GetObjectKind().GroupVersionKind()` is (mostly) empty due to [kubernetes/client-go#541](https://www.github.com/kubernetes/client-go/issues/541) so I switched to printing type of the object.